skactiveml/pool/regression/_greedy_sampling.py

```diff
@@ -0,0 +1,342 @@
+import numpy as np
+from sklearn import clone
+from sklearn.metrics import pairwise_distances
+
+from skactiveml.base import (
+    SingleAnnotatorPoolQueryStrategy,
+    SkactivemlRegressor,
+)
+from skactiveml.utils import (
+    rand_argmax,
+    labeled_indices,
+    MISSING_LABEL,
+    is_labeled,
+    check_type,
+    check_scalar,
+)
+
+
+class GreedySamplingX(SingleAnnotatorPoolQueryStrategy):
+    """Greedy Sampling on the feature space.
+
+    This class implements greedy sampling on the feature space.
+
+    Parameters
+    ----------
+    metric: str, optional (default=None)
+        Metric used for calculating the distances of points in the feature
+        space must be a valid argument for `sklearn.metrics.pairwise_distances`
+        argument `metric`.
+    missing_label : scalar or string or np.nan or None, default=np.nan
```

Comment:
I will not comment every one of them. Please see my comment above and , eventually, update all docs.